docs/en/integrations/data-ingestion/apache-spark/spark-native-connector.md

```diff
@@ -0,0 +1,583 @@
+  <groupId>com.clickhouse.spark</groupId>
+  <artifactId>clickhouse-spark-runtime-{{ spark_binary_version }}_{{ scala_binary_version }}</artifactId>
+  <version>{{ stable_version }}</version>
+</dependency>
+<dependency>
+  <groupId>com.clickhouse</groupId>
+  <artifactId>clickhouse-jdbc</artifactId>
+  <classifier>all</classifier>
+  <version>{{ clickhouse_jdbc_version }}</version>
+  <exclusions>
+    <exclusion>
+      <groupId>*</groupId>
+      <artifactId>*</artifactId>
+    </exclusion>
+  </exclusions>
+</dependency>
+```
+
+Add the following repository if you want to use SNAPSHOT version.
+
+```
+<repositories>
+  <repository>
+    <id>sonatype-oss-snapshots</id>
+    <name>Sonatype OSS Snapshots Repository</name>
+    <url>https://s01.oss.sonatype.org/content/repositories/snapshots</url>
+  </repository>
+</repositories>
+```
+</TabItem>
+<TabItem value="Gradle" label="Gradle">
+
+```gradle
+dependencies {
+  implementation("com.clickhouse.spark:clickhouse-spark-runtime-{{ spark_binary_version }}_{{ scala_binary_version }}:{{ stable_version }}")
+  implementation("com.clickhouse:clickhouse-jdbc:{{ clickhouse_jdbc_version }}:all") { transitive = false }
+}
+```
+
+Add the following repository if you want to use the SNAPSHOT version:
+
+```gradle
+repositries {
+  maven { url = "https://s01.oss.sonatype.org/content/repositories/snapshots" }
+}
+```
+</TabItem>
+<TabItem value="Spark SQL/Shell CLI" label="Spark SQL/Shell CLI">
+
+When working with Spark's shell options (Spark SQL CLI, Spark Shell CLI, Spark Submit command), the dependencies can be registered by passing the required jars:
+```text
+$SPARK_HOME/bin/spark-sql \
+  --jars /path/clickhouse-spark-runtime-{{ spark_binary_version }}_{{ scala_binary_version }}:{{ stable_version }}.jar,/path/clickhouse-jdbc-{{ clickhouse_jdbc_version }}-all.jar
+```
+If you want to avoid copying the JARs to your Spark client node, you can use the following instead:
+```text
+  --repositories https://{maven-cental-mirror or private-nexus-repo} \
+  --packages com.clickhouse.spark:clickhouse-spark-runtime-{{ spark_binary_version }}_{{ scala_binary_version }}:{{ stable_version }},com.clickhouse:clickhouse-jdbc:{{ clickhouse_jdbc_version }}:all
+```
+
+Note: For SQL-only use cases, [Apache Kyuubi](https://github.com/apache/kyuubi) is recommended
+for production.
+
+</TabItem>
+</Tabs>
+
+### Download The Library
+
+The name pattern of the binary JAR is:
+
+```
+clickhouse-spark-runtime-${spark_binary_version}_${scala_binary_version}-${version}.jar
+```
+
+You can find all available released JARs
+in the [Maven Central Repository](https://repo1.maven.org/maven2/com/clickhouse/spark/)
+and all daily build SNAPSHOT JARs
+in the [Sonatype OSS Snapshots Repository](https://s01.oss.sonatype.org/content/repositories/snapshots/com/clickhouse/).
+
+:::important
+It's essential to include the [clickhouse-jdbc JAR](https://mvnrepository.com/artifact/com.clickhouse/clickhouse-jdbc) with the "all" classifier,
```

Comment:
Maybe we need to sharpen the sentence "With Correct `clickhouse-jdbc` according to the matrix that we have above"